wazo_webhookd/controller.py

```diff
@@ -35,6 +37,11 @@ def __init__(self, config):
         ]
 
         self._auth_client = AuthClient(**config['auth'])
+        self._token_renewer = TokenRenewer(self._auth_client)
+        if not config['auth'].get('master_tenant_uuid'):
+            self._token_renewer.subscribe_to_next_token_details_change(
+                auth.init_master_tenant
+            )
```

Comment:
Since your token renewer is never used after the first token, you should not leave this thread running for nothing.
Example to stop it: https://github.com/wazo-platform/wazo-amid/blob/master/wazo_amid/controller.py#L49-L51
`subscribe_to_next_token_details_change` are executed sequentially